server/src/main/java/org/apache/cassandra/sidecar/routes/AbstractHandler.java

```diff
@@ -290,22 +292,31 @@ private Name tableName(RoutingContext context, boolean required)
      * Given a combined host address like 127.0.0.1:9042 or [2001:db8:0:0:0:ff00:42:8329]:9042, this method
      * removes port information and returns 127.0.0.1 or 2001:db8:0:0:0:ff00:42:8329.
      *
-     * @param address ip address
+     * @param request http server request
      * @return host address without port information
+     * @throws NoSuchCassandraInstanceException thrown when input address is null
      */
-    public static String extractHostAddressWithoutPort(String address)
+    @NotNull
+    public static String extractHostAddressWithoutPort(HttpServerRequest request) throws NoSuchCassandraInstanceException
     {
-        if (address.contains(":"))
+        String host = request.host();
+        if (host == null)
+        {
+            throw new NoSuchCassandraInstanceException("No such Cassandra instance when Host header is absent");
```

Comment:
Nit: If host is not set in request, should we send back `Bad Request: Missing host header` instead of `404`

Reply:
+10000
Very good point